internal/controller/deployment_controller.go

```diff
@@ -403,6 +409,60 @@ func (r *DeploymentReconciler) Delete(ctx context.Context, l logr.Logger, deploy
 	return ctrl.Result{RequeueAfter: 10 * time.Second}, nil
 }
 
+func (r *DeploymentReconciler) keepOrReleaseAWSCluster(ctx context.Context, l logr.Logger, namespace, clusterName string) error {
+	found, err := r.machinesAvailable(ctx, namespace, clusterName)
+	if err != nil {
+		return err
+	}
+
+	return r.updateAWSCluster(ctx, l, namespace, clusterName, found)
+}
+
+func (r *DeploymentReconciler) updateAWSCluster(ctx context.Context, l logr.Logger, namespace, clusterName string, disableDeletion bool) error {
+	cluster := &capiaws.AWSCluster{}
+	err := r.Get(ctx, types.NamespacedName{
+		Name:      clusterName,
+		Namespace: namespace,
+	}, cluster)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	finalizersUpdated := false
+	if disableDeletion {
+		l.Info("Ensure we have cannot stop AWSCluster", "finalizer", hmc.BlockingFinalizer)
+		finalizersUpdated = controllerutil.AddFinalizer(cluster, hmc.BlockingFinalizer)
```

Comment:
Can we consider the option to set the finalizer initially, when creating the `AWSCluster` object? I mean, add the finalizer to the `AWSCluster` manifest in the templates?